sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/service/servicebus/factory/ServiceBusReceiverClientBuilderFactory.java

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.service.servicebus.factory;
+
+import com.azure.messaging.servicebus.ServiceBusClientBuilder;
+import com.azure.spring.service.core.PropertyMapper;
+import com.azure.spring.service.servicebus.properties.ServiceBusConsumerDescriptor;
+import com.azure.spring.service.servicebus.properties.ServiceBusEntityType;
+import org.springframework.util.Assert;
+
+import static com.azure.spring.service.servicebus.properties.ServiceBusEntityType.TOPIC;
+
+/**
+ * Service Bus client builder factory, it builds the {@link ServiceBusClientBuilder.ServiceBusReceiverClientBuilder}.
+ */
+public class ServiceBusReceiverClientBuilderFactory
+    extends AbstractServiceBusSubClientBuilderFactory<ServiceBusClientBuilder.ServiceBusReceiverClientBuilder,
+    ServiceBusConsumerDescriptor> {
+
+    private final ServiceBusConsumerDescriptor consumerProperties;
+
+    public ServiceBusReceiverClientBuilderFactory(ServiceBusConsumerDescriptor consumerDescriptor) {
+        this(null, consumerDescriptor);
```

Comment:
what happends if this is null?

Reply:
A another ServiceBusClientBuilder will be created and this is used in the case of a dedicated connection of service bus entity need to be used, so we need a new ServiceBusClientBuilder .